gu-net/src/rpc/peer.rs

```diff
@@ -130,3 +177,91 @@ impl Handler<GetPeer> for PeerManager {
         MessageResult(self.peers.get(&msg.0).cloned())
     }
 }
+
+// Tag management
+pub struct AddTags {
+    pub node: NodeId,
+    pub tags: Tags,
+}
+
+impl Message for AddTags {
+    type Result = Option<()>;
+}
+
+impl Handler<AddTags> for PeerManager {
+    type Result = Option<()>;
+    fn handle(&mut self, msg: AddTags, ctx: &mut Self::Context) -> Self::Result {
+        let mut peer = self.peers.get_mut(&msg.node)?;
+        let mut tags = &mut peer.tags;
+        tags.extend(msg.tags);
+        Some(())
```

Comment:
We can't change the tags of a disconnected provider.

Reply:
Fixed.